src/profile-logic/tracks.js

```diff
@@ -115,12 +115,19 @@ function _getDefaultLocalTrackOrder(tracks: LocalTrack[], profile: ?Profile) {
 
     // If the tracks are both threads, sort them by thread name, and then by
     // creation time if they have the same name.
-    if (tracks[a].type === 'thread' && tracks[b].type === 'thread' && profile) {
+    if (
+      (tracks[a].type === 'thread' || tracks[a].type === 'marker') &&
+      (tracks[b].type === 'thread' || tracks[b].type === 'marker') &&
+      profile
+    ) {
       const idxA = tracks[a].threadIndex;
       const idxB = tracks[b].threadIndex;
       if (idxA === undefined || idxB === undefined) {
         return -1;
       }
+      if (profile && profile.meta.disableThreadOrdering) {
+        return idxA - idxB;
+      }
```

Comment:
TBH I expected this to be before the sort. Something like this:
```js
const trackOrder = tracks.map((_, index) => index);
if (profile && profile.meta.disableThreadOrdering) {
  return trackOrder;
}
... carry on with the sorting
```

Therefore I'm confused with the goal for this flag. It doesn't seem to do what it says.

My 2 cents: I think that a simple flag enable/disable sorting isn't what we want, because this is too much black or white. What do you think about having `LOCAL_TRACK_DISPLAY_ORDER` being specified in the profile instead?

Taking one step back, I think I'd like to understand your use case here. Can you explain it more?

Reply:
`disableThreadOrdering` == use the order in which the threads are defined in the profile file (no thread specific ordering).

Reply:
Please comeback to me if this is unclear.